### PR TITLE
Fix condition to determine if the date field's value is an array

### PR DESCRIPTION
**Pull request by rupal-bq**
### Description

- Fix condition to determine if the date field's value is an array
- Add loose flag to OSD bootstrap to fix
```
ERROR [single_version_dependencies] Multiple version ranges for the same dependency
      were found declared across different package.json files. Please consolidate
      those to match across all package.json files. Different versions for the
      same dependency is not supported.

      If you have questions about this please reach out to the operations team.

      The conflicting dependencies are:

        cypress
          9.5.4 => opensearch-dashboards
          ^13.6.0 => reports-dashboards
error Command failed with exit code 1.
```

### Issues Resolved
https://github.com/opensearch-project/reporting/issues/932
https://github.com/opensearch-project/dashboards-reporting/issues/257

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by YANG-DB — APPROVED** [review]

**Review by ps48 — APPROVED** [review]

File: .github/workflows/cypress-e2e-reporting-test.yml [hunk 1]
```diff
@@ -150,7 +150,7 @@ jobs:
 
       - name: Run Cypress tests
         run: |
-          yarn cypress:run --browser chrome --headless --spec '.cypress/integration/*'
+          yarn cypress:run --browser chrome --headless
```
Comment (RyanL1997) [inline .github/workflows/cypress-e2e-reporting-test.yml:153]: Oops good catch on this, seems like we don't need to specify the `spec` for this directory setup. Thanks!

File: DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -27,7 +27,7 @@ another JDK installation, e.g. `RUNTIME_JAVA_HOME=/usr/lib/jvm/jdk-8`.
    cd plugins
    git clone https://github.com/opensearch-project/dashboards-reporting.git
    ```
-1. Run `yarn osd bootstrap` inside `OpenSearch-Dashboards/plugins/dashboards-reporting`.
+1. Run `yarn osd bootstrap --single-version=loose` inside `OpenSearch-Dashboards/plugins/dashboards-reporting`.
```
Comment (RyanL1997) [inline DEVELOPER_GUIDE.md:30]: Nice!

File: server/routes/utils/dataReportHelpers.ts [hunk 3]
```diff
@@ -145,8 +145,8 @@ export const getOpenSearchData = (
             if (typeof dateValue === 'string') {
               data._source[keys] = moment.utc(dateValue).tz(timezone).format(dateFormat);
             } else if (
-              fieldDateValue.length !== 0 &&
-              fieldDateValue instanceof Array
+              dateValue.length !== 0 &&
+              dateValue instanceof Array
             ) {
               fieldDateValue.forEach((element, index) => {
```
Comment (RyanL1997) [inline server/routes/utils/dataReportHelpers.ts:151]: Just for my knowledge, why whats the difference between `fieldDateValue ` and `dateValue ` above. Cuz I saw the change was trying to switch to `dateValue `.

**Review by RyanL1997 — APPROVED** [review]
Thanks for putting this together @rupal-bq! I just left some minor reviews

File: server/routes/utils/dataReportHelpers.ts (hunk 3 above)
Comment (rupal-bq) [reply server/routes/utils/dataReportHelpers.ts:151]: Originally the variable was `dateValue` for this condition but was mistakenly changed in different PR which caused `TypeError: Cannot create property '0' on number '1700813322772'`. Reverting to fix the issue.

**Pull request merged**